airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -176,14 +175,14 @@ private static GoogleCredentials getServiceAccountCredentials(final JsonNode con
     if (!BigQueryUtils.isUsingJsonCredentials(config)) {
       LOGGER.info("No service account key json is provided. It is required if you are using Airbyte cloud.");
       LOGGER.info("Using the default service account credential from environment.");
-      return ServiceAccountCredentials.getApplicationDefault();
+      return GoogleCredentials.getApplicationDefault();
```

Comment:
according to the Google OAuth2 documentation in case the path to the file with credentials is set in the env variable `GOOGLE_APPLICATION_CREDENTIALS`, then GoogleCredentials will be used by default

Reply:
Yes, but GOOGLE_APPLICATION_CREDENTIALS do not passthrough to docker env. Even if you add it to docker-compose, still it is the path only, and you would need to mount the volume containing the file. This method only works if you are inside cloud shell for example, as on cloud shell there is a endpoint that default credentials will use to get the credentials (there is no GOOGLE_APPLICATION_CREDENTIALS set by default on cloud shell)

Reply:
> Google Cloud Client Libraries use a library called Application Default Credentials (ADC) to automatically find your service account credentials. ADC looks for service account credentials in the following order:
>
>If the environment variable GOOGLE_APPLICATION_CREDENTIALS is set, ADC uses the service account key or configuration file that the variable points to.
>
>If the environment variable GOOGLE_APPLICATION_CREDENTIALS isn't set, ADC uses the service account that is attached to the resource that is running your code.
>
>This service account might be a [default service account](https://cloud.google.com/iam/docs/service-accounts#default) provided by Compute Engine, Google Kubernetes Engine, App Engine, Cloud Run, or Cloud Functions. It might also be a [user-managed service account](https://cloud.google.com/iam/docs/service-accounts#user-managed) that you created.
>
>If ADC can't use any of the above credentials, an error occurs.

https://cloud.google.com/docs/authentication/production

Reply:
So, to use `GOOGLE_APPLICATION_CREDENTIALS`, the worker would need access to the file, not only the `GOOGLE_APPLICATION_CREDENTIALS` variable, which is not feasible, as we would need to mount the volume containing the file that `GOOGLE_APPLICATION_CREDENTIALS` points too

Reply:
hi @marcelopio , thank you for providing more context on the issue
`ServiceAccountCredentials` is extending `GoogleCredentials` and doesn't override the `getApplicationDefault()` method, which means that it doing the same stuff, so I don't see how it will fix your issue. Have you checked that it actually makes a difference and resolves the problem?

Reply:
Yes, it will enable me to get the json generated by [gcloud auth application-default login](https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login) and load into Airbyte. The problem with current implementation is that it only accepts Service Account Jsons. So the part I want to really change is [line 185](https://github.com/airbytehq/airbyte/blob/9ea4a1d4f8dff7905230c972dfcce8d8ebd23b84/airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java#L185) which enables any valid google json (including oauth2 and other credentials)